internal/ext/olm/constraints_test.go

```diff
@@ -0,0 +1,499 @@
+package olm
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"sort"
+	"testing"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"github.com/operator-framework/deppy/internal/entitysource"
+	"github.com/operator-framework/deppy/internal/sat"
+)
+
+func TestConstraints(t *testing.T) {
+	RegisterFailHandler(Fail)
+	RunSpecs(t, "Constraints Suite")
+}
+
+// MockQuerier type to mock the entity querier
+type MockQuerier struct {
+	returnError         bool
+	errorString         string
+	entityListReturn    entitysource.EntityList
+	entityListMapReturn entitysource.EntityListMap
+}
+
+func (t MockQuerier) Get(ctx context.Context, id entitysource.EntityID) *entitysource.Entity {
+	return &entitysource.Entity{}
+}
+func (t MockQuerier) Filter(ctx context.Context, filter entitysource.Predicate) (entitysource.EntityList, error) {
+	if t.returnError {
+		return nil, errors.New(t.errorString)
+	}
+	return t.entityListReturn, nil
+}
+func (t MockQuerier) GroupBy(ctx context.Context, id entitysource.GroupByFunction) (entitysource.EntityListMap, error) {
+	if t.returnError {
+		return nil, errors.New(t.errorString)
+	}
+	return t.entityListMapReturn, nil
+}
+func (t MockQuerier) Iterate(ctx context.Context, id entitysource.IteratorFunction) error {
+	if t.returnError {
+		return errors.New(t.errorString)
+	}
+	return nil
+}
+
+var _ = Describe("Constraints", func() {
+	Context("requirePackage", func() {
+		Describe("GetVariables", func() {
+			var (
+				ctx         context.Context
+				reqPkg      requirePackage
+				mockQuerier MockQuerier
+			)
+			BeforeEach(func() {
+				ctx = context.Background()
+				reqPkg = requirePackage{
+					packageName:  "cool-package-1",
+					versionRange: "1.0-2.0",
+					channel:      "my-channel",
+				}
+				mockQuerier = MockQuerier{
+					returnError: false,
+				}
+			})
+			It("returns a slice of sat.Variable containing requirement on entity", func() {
+				mockQuerier.entityListReturn = entitysource.EntityList{
+					*entitysource.NewEntity("cool-package-2-entity", map[string]string{
+						propertyOLMPackageName: "cool-package-2",
+					}),
+				}
+				vars, err := reqPkg.GetVariables(ctx, mockQuerier)
+				expectedIdentifier := fmt.Sprintf("require-%s-%s-%s", reqPkg.packageName, reqPkg.versionRange, reqPkg.channel)
+				Expect(err).NotTo(HaveOccurred())
+				Expect(vars).Should(HaveLen(1))
+				Expect(vars[0].Identifier().String()).To(Equal(expectedIdentifier))
+				Expect(vars[0].Constraints()).Should(HaveLen(2))
+				Expect(vars[0].Constraints()[0].String(sat.IdentifierFromString(expectedIdentifier))).To(
```

Comment:
suggestion: aki found some nice standard gomega matchers for lists and maps, [e.g.](https://github.com/operator-framework/deppy/pull/70/files#diff-b616f658d565fcb1ee81cf023b3570b56e7625eba9670a598b9e546027c6c75eR112)